cmake/Corrosion.cmake

```diff
@@ -1397,3 +1397,74 @@ function(corrosion_install)
         message(FATAL_ERROR "install(EXPORT ...) not yet implemented")
     endif()
 endfunction()
+
+# Create a library target that will run cxxbridge on the given files on a crate
+# that is assumed to contain a module with #[cxx::bridge]
+# Call with corrosion_cxxbridge(<target> CRATE <crate> PATH <tomlPath> FILES [<filePath>, ...])
+# <target> is the name of the target created that will host the cxx interface
+# CRATE <crate> is the target name of the specific target generated by corrosion_import_crate
+# PATH <tomlPath> is the relative path from where this function is called to Cargo.toml of <crate>
+# FILES [<filePath, ...] is the relative paths to files with #[cxx::bridge]. These are relative to <tomlPath>
+# 
+# Example: corrosion_cxxbridge(myCxxTarget CRATE someCrate PATH someCratePath FILES src/lib.rs)
+function(corrosion_cxxbridge target)
+    # This function is based on the details given by https://cxx.rs/build/other.html#producing-the-generated-code
+    # corrosion_cxxbridge requires the cxxbridge executable to be installed
+    # run `cargo install cxxbridge-cmd` if missing
+    find_program(CXXBRIDGE cxxbridge PATHS "$ENV{HOME}/.cargo/bin/" REQUIRED)
+
+    set(OPTIONS)
+    set(ONE_VALUE_KEYWORDS CRATE PATH)
+    set(MULTI_VALUE_KEYWORDS FILES)
+    cmake_parse_arguments(_arg "${OPTIONS}" "${ONE_VALUE_KEYWORDS}" "${MULTI_VALUE_KEYWORDS}" ${ARGN})
+
+    if("${_arg_CRATE}" STREQUAL "")
+        message(FATAL_ERROR "corrosion_cxxbridge called without a crate")
+    endif()
+
+    if("${_arg_PATH}" STREQUAL "")
+        message(FATAL_ERROR "corrosion_cxxbridge called without path to folder of Cargo.toml of target ${_arg_CRATE}")
+    endif()
+
+    if(NOT EXISTS "${CMAKE_CURRENT_LIST_DIR}/${_arg_PATH}/Cargo.toml")
+        message(FATAL_ERROR "The path ${CMAKE_CURRENT_LIST_DIR}/${_arg_PATH} does not contain a Cargo.toml")
+    endif()
+
+    if("${_arg_FILES}" STREQUAL "")
+        message(FATAL_ERROR "Must supply at least the path of a single file to be bridged, defined as relative path from Cargo.toml")
+    endif()
+
+    set(crate_path ${_arg_PATH})
+
+    add_library(${target})
+    target_include_directories(${target} PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/cxxgenerated)
+    target_link_libraries(${target} PRIVATE ${_arg_CRATE})
+
+    set(cxx_generated_dir ${CMAKE_CURRENT_BINARY_DIR}/cxxgenerated/${target})
+
+    foreach(file ${_arg_FILES})
+        get_filename_component(filename ${file} NAME_WE)
+        set(cxx_header ${filename}.h)
+        set(cxx_source ${filename}.cpp)
+
+        add_custom_command(
+            TARGET cargo-build_${_arg_CRATE}
+            POST_BUILD
+            COMMAND
+                "${CMAKE_COMMAND}" -E make_directory ${cxx_generated_dir}
+            COMMAND
+                ${CXXBRIDGE} ${CMAKE_CURRENT_LIST_DIR}/${crate_path}/${file} --header > ${cxx_generated_dir}/${cxx_header}
```

Comment:
Does this even need to depend on `cargo-build`? The documentation does not say that the tool has to run before/after cargo build, so maybe it can be run independently?

Reply:
Yeah, that is true! It can be run whenever, and since it links in the imported rust target, it will regardless be built correctly, will fix!

Reply:
Update on this. If we depend on `TARGET cargo-build_<importedtarget>` then we will always rerun the cxxbridge generation if this target changes.
It might run a bit too optimistically, but again should _always_ run when it needs to.
Im guessing this would be the preferable case?

Reply:
Yes, I think always rerunning is the only sensible thing for now. We can still optimize it later.
FYI, The cargo-build target is a custom command, so it is always out of date and rerun.

Reply:
Ah, yeah, Ill change it to the actual target, thanks!

Reply:
Can't actually use that target since its an interface target. Depending on the exact rust source file instead.

Reply:
>  Ah, yeah, Ill change it to the actual target, thanks!

I think you may have misunderstood. I meant since the `cargo-build_xxx` target is a custom target and thus always out of date, the `POST_BUILD` step would also always be out of date. So the original approach was perfectly fine and the more conservative approach compared to what you changed to now.

Reply:
Yup I definitely misunderstood, and agreed. I'll keep it as it originally was. It's the most safe case for now.